### PR TITLE
yajl: try to fix compilation with MSVC

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: subprojects/packagefiles/yajl/meson.build [hunk 1]
```diff
@@ -1,4 +1,9 @@
 project('YetAnotherJSONParser', 'c', version: '2.1.0')
+
+if get_option('buildtype') != 'static'
```
Comment (xclaesse) [inline subprojects/packagefiles/yajl/meson.build:3]: Do you mean `default_library`?

Comment (neheb) [reply subprojects/packagefiles/yajl/meson.build:3]: Yep. Turns out there are 12 more places in here with the same mistake.

Comment (neheb) [reply subprojects/packagefiles/yajl/meson.build]: Fixed.

**Comment by neheb** [comment]
Anything else?

**Comment by xclaesse** [comment]
@neheb sorry I'm late at reviewing this. Could you update the PR to remove it from broken_windows list please? Other than that LGTM.

**Comment by neheb** [comment]
Done.

**Pull request merged**